### PR TITLE
Create `CONTRIBUTING.md`

### LINKED ISSUE #566: add CONTRIBUTE.md

### PR DESCRIPTION
**Pull request by kysrpex**
Create the file `CONTRIBUTING.md`, which provides guidelines on how to contribute to SimPhoNy.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[simphony@iwm.fraunhofer.de](mailto:simphony@iwm.fraunhofer.de).
+All complaints will be reviewed and investigated promptly and fairly.
+
+All community leaders are obligated to respect the privacy and security of the
+reporter of any incident.
+
+## Enforcement Guidelines
```
Comment (kysrpex) [inline CODE_OF_CONDUCT.md:69]: Revise that the consequences of the actions are adequate

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Estimation of the effort that implementing such enhancement or feature might involve. Technologies that could enable the implementation. This information helps the maintainers decide whether to implement a feature or enhancement, and to estimate the timeframe for a potential implementation.
+
+After this discussion phase, when the maintainers deem it appropriate, the discussion on the forum can be closed and an issue referencing it can be created on the [issue board](https://github.com/simphony/osp-core/issues?q=is%3Aissue+sort%3Aupdated-desc+). The goal of this procedure is to separate discussion and design from implementation efforts. The issue is now ready to be worked on. The [_code section_](#Code) explains how to contribute to solve an issue.
+
+## Code
+
+Code contributions are generally aimed at fixing bugs and implementing enhancements or new features. The way to contribute code to the project are [pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests).
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
Code contributions are generally aimed at fixing bugs and implementing enhancements or new features. The way to contribute code to the project are [pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) (PR).
```
So it is clear when you use "PR" from now on

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+If you are unsure about where to start, check the issues tagged as [_newcomer task_](https://github.com/simphony/osp-core/issues?q=label%3A%22%F0%9F%91%A9%E2%80%8D%F0%9F%8E%93+newcomer+task%22+sort%3Acomments-desc). Such issues should either be simple to solve, or be a good entry point for understanding the codebase.
+
+If you have a bugfix or have a feature/enhancement implementation that you want to contribute, please read [this page](https://simphony.readthedocs.io/en/latest/contribute.html) to understand how the code is organized, what is the meaning of the different git branches on the repository, and what automatic checks your code will have to pass in order to be accepted. After that, just fork the repository and make a new pull request.
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
If you have a bugfix or a feature/enhancement implementation that you want to contribute, please read [this page](https://simphony.readthedocs.io/en/latest/contribute.html) to understand how the code is organized, what the meaning of the different git branches on the repository is, and what automatic checks your code will have to pass in order to be accepted. After that, just fork the repository and make a new pull request.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+2. Check the [Q&A](https://github.com/simphony/osp-core/discussions/categories/q-a) section of the [discussions page](https://github.com/simphony/osp-core/discussions). Someone else might have already experienced the same issue and got an answer on how to solve it.
+3. Search for similar issues on the [issue board](https://github.com/simphony/osp-core/issues?q=is%3Aissue+sort%3Aupdated-desc+). The problem may already have been reported. If you are having issues while using a specific SimPhoNy wrapper, beware that each SimPhoNy Wrapper has its own issue board.
+
+#### Submitting a bug report
+
+Bug reports should be submitted on the [SimPhoNy issue board](https://github.com/simphony/osp-core/issues?q=is%3Aissue+sort%3Aupdated-desc+).  If you suspect that the issue has its origin in the code of a specific wrapper, rather than in SimPhoNy itself, visit the issue board of the wrapper instead. To create a new report, click the green [_"New"_](https://github.com/simphony/osp-core/issues/new/choose) button.
+
+- [ ] Choose a meaningful title.
+- [ ] Describing the problem in few words.
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
- [ ] Describe the problem in few words.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Specify which version of the `simphony-osp` is affected. Use `pip show simphony-osp` to find out which version do you have installed. If the problem involves a wrapper, provide also the version of the wrapper.
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
- [ ] Specify which version of `simphony-osp` is affected. Use `pip show simphony-osp` to find out which version you have installed. If the problem involves a wrapper, provide also the version of the wrapper.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+- [ ] Explain how to reproduce the problem, step-by-step, and include a **minimal reproducible example**. A _minimal reproducible example_ is a code snippet where the issue can be observed. Include any additional files (e.g. an ontology file) that may be needed to execute the example. Skip the example _only if it is very difficult to provide it_.
+- [ ] If the issue involves a crash or an exception, include the error message.
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
- [ ] If the issue involves a crash or an exception, include the full error message (with the stacktrace).
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+[Follow this link](https://github.com/simphony/osp-core/issues/740#issue-1107800007) to see an example of an accurate bug report. Providing a good bug report facilitates the work of the maintainers and enables them to solve the issue faster.
+
+### Proposing enhancements or suggesting new features
+
+Enhancements or new features should first be proposed on the [_"Ideas"_ section of the forum](https://github.com/simphony/osp-core/discussions/categories/ideas).
+
+When proposing, discussing, and designing a feature or enhancement on the forum, always aim to steer the discussion in a direction such that the following points are addressed:
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
When proposing, discussing, and designing a feature or enhancement on the forum, always aim to address the following points:
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+After you submit your pull request, a maintainer will review it. It is possible that additional work is needed before the maintainer can accept the PR. To make things smoother, please consider [allowing us to directly edit your PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so that we can perform minor edits without having to wait for your feedback.
+
+Of course, even if you are not a maintainer, you are also very welcome to comment on the pull requests submitted by other community members and give feedback to them.
+
+## Documentation
+
+If you feel that the [SimPhoNy documentation](https://simphony.readthedocs.io) it is difficult to understand, we welcome your feedback and contributions! In general, changes to the documentation should be first proposed on the [_"Ideas"_ section of the forum](https://github.com/simphony/osp-core/discussions/categories/ideas).
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
If you feel that the [SimPhoNy documentation](https://simphony.readthedocs.io) is difficult to understand, we welcome your feedback and contributions! In general, changes to the documentation should be first proposed on the [_"Ideas"_ section of the forum](https://github.com/simphony/osp-core/discussions/categories/ideas).
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+However, in certain cases, a change in the documentation does not need to be discussed. For example, if you find a typo you can directly [create an issue](https://github.com/simphony/docs/issues/new/choose) or even a [pull request](https://github.com/simphony/docs/compare). The same applies if the documentation needs to be changed due to the implementation of enhancements or new features for SimPhoNy in a [code contribution](#code).
+
+When proposing and discussing a change on the documentation, always aim to steer the discussion in a direction such that the following points are addressed:
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
When proposing and discussing a change on the documentation, always aim to address the following points:
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+- [ ] Motivation for the change or addition.
+- [ ] A brief description, in few sentences, of the proposed changes.
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
- [ ] A brief description of the proposed changes.
```

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (pablo-de-andres) [reply CODE_OF_CONDUCT.md:69]: I went through it, makes sense to me.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
+# Contributing to SimPhoNy
+
+Thank you for considering to improve SimPhoNy. We welcome your contribution!
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
Thank you for considering to improve SimPhoNy. We welcome your input!
```
(A lot of mentions to contribution)

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+  </table>
+</div>
+
+Remember that you must adhere to the [Contributor Covenant Code of Conduct](https://github.com/simphony/osp-core/blob/master/CODE_OF_CONDUCT.md) if you decide to make a contribution.
+
+## Issues
+
+### Reporting bugs
+
+When you find that something is not working, you can submit a bug report. This section provides guidelines on how to submit your report in a way that helps maintainers understand the problem and reproduce it. Submitting a good bug report is a key to have issues solved quickly.
```
Comment (pablo-de-andres) [inline CONTRIBUTING.md]:
```suggestion
When you find something that is not working as it should, you can submit a bug report. This section provides guidelines on how to make your report in a way that helps maintainers understand the problem and reproduce it. Submitting a good bug report is a key to have issues solved quickly.
```

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (kysrpex) [reply CODE_OF_CONDUCT.md:69]: Thanks. They also seem to make sense to me.

File: CONTRIBUTING.md (hunk 2 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 3 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 4 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 5 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 6 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 7 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 8 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 9 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 10 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 11 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: Even though there are lots of mentions, contribution still sounds better than input to me here.

File: CONTRIBUTING.md (hunk 12 above)
Comment (kysrpex) [reply CONTRIBUTING.md]: I preferred to use the word _write_ instead of _make_.

**Comment by kysrpex** [comment]
@pablo-de-andres There are still two open conversations: [this one ](https://github.com/simphony/osp-core/pull/799#discussion_r913605343) and [this one](https://github.com/simphony/osp-core/pull/799#discussion_r913610296) that you can object to or agree with.

**Pull request merged**